src/ResourceManager/KeyVault/Commands.KeyVault/Commands/AddAzureKeyVaultCertificateContact.cs

```
@@ -0,0 +1,115 @@
+        public string VaultName { get; set; }
+
+        /// <summary>
+        /// EmailAddress
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   ParameterSetName = AddParameterSet,
+                   Position = 1,
+                   ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "Specifies the email address of the contact.")]
+        [ValidateNotNullOrEmpty]
+        public string EmailAddress { get; set; }
+
+        /// <summary>
+        /// PassThru parameter
+        /// </summary>
+        [Parameter(HelpMessage = "This cmdlet does not return an object by default. If this switch is specified, it returns the contact object.")]
+        public SwitchParameter PassThru { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            Contacts existingContacts;
+
+            try
+            {
+                existingContacts = this.DataServiceClient.GetCertificateContacts(VaultName);
+            }
+            catch(KeyVaultErrorException exception)
+            {
+                if (exception.Response.StatusCode != System.Net.HttpStatusCode.NotFound)
+                {
+                    throw;
+                }
+
+                existingContacts = null;
+            }
+
+            List<Contact> newContactList;
+
+            if (existingContacts == null ||
+                existingContacts.ContactList == null)
+            {
+                newContactList = new List<Contact>();
+            }
+            else
+            {
+                newContactList = new List<Contact>(existingContacts.ContactList);
+            }
+
+            if (newContactList.FindIndex(
+                contact => (string.Compare(contact.EmailAddress, EmailAddress, StringComparison.OrdinalIgnoreCase) == 0)) != -1)
+            {
+                throw new ArgumentException(string.Format(CultureInfo.InvariantCulture, "Provided email address '{0}' already exists.", EmailAddress));
```

Do we want to throw an exception or just return the result?